fapolicy_analyzer/ui/fapd_manager.py

```diff
@@ -0,0 +1,215 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import subprocess
+
+
+from datetime import datetime as DT
+from enum import Enum
+from fapolicy_analyzer import Handle
+from threading import Lock
+from time import time, sleep
+
+
+class ServiceStatus(Enum):
+    FALSE = False
+    TRUE = True
+    UNKNOWN = None
+
+
+class FapdMode(Enum):
+    DISABLED = 0
+    ONLINE = 1
+    PROFILING = 2
+
+
+class FapdManager():
+    def __init__(self, bSuFapdControl=True):
+        logging.debug("FapdManager::__init__(self)")
+        self._fapd_status = ServiceStatus.UNKNOWN
+        self._fapd_monitoring = False
+        self._fapd_ref = Handle("fapolicyd")
+        self._fapd_profiler_pid = None
+        self._fapd_control_enabled = bSuFapdControl
+        self._fapd_lock = Lock()
+        self.mode = FapdMode.DISABLED
+        self.procProfile = None
+        self._fapd_profiling_timestamp = None
+
+        # Verify that fapolicyd is intalled, if so initialize fapd status
+        self.initial_daemon_status()
+
+        # ToDo: Verify that FAPD_LOGPATH works in the pkexec environment
+        if os.environ.get("FAPD_LOGPATH"):
+            self.fapd_profiling_stdout = os.environ.get("FAPD_LOGPATH") + ".stdout"
+            self.fapd_profiling_stderr = os.environ.get("FAPD_LOGPATH") + ".stderr"
+        else:
+            self.fapd_profiling_stdout = None
+            self.fapd_profiling_stderr = None
+
+    # ############### Public Interface ###############
+    def start_online_fapd(self):
```

Comment:
The `_fapd` suffix on this and the following methods seems redundant.

Reply:
Removed `_fapd` suffix on public functions.

Reply:
Done; Pushed.